File: englising-be/src/main/java/org/englising/com/englisingbe/singleplay/entity/SinglePlay.java
package org.englising.com.englisingbe.singleplay.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.englising.com.englisingbe.music.entity.Track;
import org.englising.com.englisingbe.user.entity.User;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "singleplay")
public class SinglePlay {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "singleplay_id")
    private Long singlePlayId;

    @ManyToOne
    @JoinColumn(name = "singleplay_level_id", referencedColumnName = "singleplay_level_id")
    private SinglePlayHint singlePlayHint;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "track_id", referencedColumnName = "track_id")
    private Track track;

    @Column(name = "score")
    private Integer score;

    @Column(name = "correct_rate")
    private Integer correctRate;

    @CreationTimestamp
    @Column(name = "created_at")
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Timestamp updatedAt;
}

File: englising-recommend/model/single_play.py
from sqlalchemy import Column, Integer, ForeignKey, String, TIMESTAMP, Boolean, SmallInteger
from sqlalchemy.sql import func
from model.base import Base

class Singleplay(Base):
    __tablename__ = 'singleplay'

    singleplay_id = Column(Integer, primary_key=True, autoincrement=True)
    singleplay_level_id = Column(SmallInteger, ForeignKey('singleplay_hint.singleplay_level_id'), nullable=False)
    user_id = Column(Integer, ForeignKey('user.user_id'), nullable=False)
    track_id = Column(Integer, ForeignKey('track.track_id'), nullable=False)
    score = Column(SmallInteger, nullable=True)
    correct_rate = Column(SmallInteger, nullable=True)
    created_at = Column(TIMESTAMP, default=func.now(), nullable=False)
    updated_at = Column(TIMESTAMP, onupdate=func.now(), nullable=True)
